### PR TITLE
add a messages tab for surfing

### PR DESCRIPTION
**Pull request by amdwit**
I had to delete the previous pr. Something bad happened while rebasing. All comments have been addressed.

### CODE REVIEW

File: app/frontend/src/features/messages/surfing/HostRequestView.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      );
+      setMessages(messages);
+    } catch (error) {
+      setError(error.message);
+    }
+    setLoading(false);
+  }, [hostRequest.hostRequestId]);
+
+  useEffect(() => {
+    fetchUsers({ userIds: [hostRequest.fromUserId, hostRequest.toUserId] });
```
Comment (lucaslcode) [inline app/frontend/src/features/messages/surfing/HostRequestView.tsx]:
```suggestion
    dispatch(fetchUsers({ userIds: [hostRequest.fromUserId, hostRequest.toUserId] }));
```

**Review by lucaslcode — CHANGES_REQUESTED** [review]

File: app/frontend/src/features/messages/surfing/HostRequestView.tsx (hunk 1 above)
Comment (amdwit) [reply app/frontend/src/features/messages/surfing/HostRequestView.tsx]: Indeed. But in this `fetchUsers` there is also a dispatch. So we need two to make it work.

File: app/frontend/src/features/messages/surfing/HostRequestView.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+      );
+      setMessages(messages);
+    } catch (error) {
+      setError(error.message);
+    }
+    setLoading(false);
+  }, [hostRequest.hostRequestId]);
+
+  useEffect(() => {
+    dispatch(fetchUsers({ userIds: [hostRequest.fromUserId, hostRequest.toUserId] }));
```
Comment (darrenvong) [inline app/frontend/src/features/messages/surfing/HostRequestView.tsx:40]: It says the build in CI is failing on this line. You'll need to `useAppDispatch` to get a reference to `dispatch`.

**Review by darrenvong — DISMISSED** [review]

File: app/frontend/src/features/messages/surfing/HostRequestView.tsx (hunk 2 above)
Comment (amdwit) [reply app/frontend/src/features/messages/surfing/HostRequestView.tsx:40]: Of course!

**Review by lucaslcode — APPROVED** [review]

**Pull request merged**